File: Assignment 1 08-01-19/complex_number.cpp
//08-01-19
#include<bits/stdc++.h>
using namespace std;
class complexx
{
	private:
		float real,imag;
	public:
		complexx()
		{
			real=0;
			imag=0;
		}
		complexx(float a,float b)
		{
			real=a;
			imag=b;
		}
		void display()
		{
		cout<<real<<"+("<<imag<<"i)"<<"\n";
		}
		
		
		void input()
		{
			cin>>real>>imag;
		}
		
		complexx operator + (complexx number)
		{
			complexx temp;
			temp.real=real+number.real;
			temp.imag=imag+number.imag;
			return temp;
		}
		complexx operator - (complexx number)
		{
			complexx temp;
			temp.real=real-number.real;
			temp.imag=imag-number.imag;
			return temp;
		}
		complexx operator * (complexx number)
		{
			complexx temp;
			temp.real=real*number.real - imag*number.imag;
			temp.imag=real*number.imag+number.real*imag;
			return temp;
		}
		complexx operator / (complexx number)
		{
			complexx temp;
			float d;
			d=pow(number.real,2)+pow(number.imag,2);
			temp.real=(real*number.real+imag*number.imag)/d;
			temp.imag=(number.real*imag-real*number.imag)/d;
			return temp;
		}
};

int main()
{
//	complexx c1(3,4),c2(4,5),c3,c4,c5,c6;
	complexx c1,c2,c3,c4,c5,c6;
	c1.input();
	c2.input();
	c1.display();
	c2.display();
	cout<<"Addition:";
	c3=c1+c2;
	c3.display();
	cout<<"Subtraction:";
	c4=c2-c1;
	c4.display();
	cout<<"Multiplication:";
	c5=c1*c2;
	c5.display();
	cout<<"Division:";
	c6=c1/c2;
	c6.display();
	cout<<endl;
	return 0;
}

File: Assignment 1 08-01-19/complex_number.py
#input should be of the form a+bj

i=complex(input())
j=complex(input())

print('number-1',i.real,i.imag)
print('number-2',j.real,j.imag)

#print(type(i)) 
#print(i)
k=i+j
print("after adding")
print(k)
k=i-j
print("after subtracting")
print(k)
k=i*j
print("after multiplying");
print(k)
k=i/j
print("after dividing");
print(k)
'''
X=[int(x) for x in input().split(' ')]
Y=[int(y) for y in input().split(' ')]

a=X[0]
b=X[1]
c=Y[0]
d=Y[1]
print("after adding")
print(str(a+c)+"+i"+str(b+d))
print("after subtracting")
print(str(a-c)+"+i"+str(b-d))


if b*d<0:
	print("after multiplying");
	print(str((a*c)-(b*d))+"+i"+str((a*d)+(b*c)));	
	
	print("after dividing");
	print(str(float(((a*c)-(b*d))/(c*c-d*d)))+"+i"+str(float(((a*d)+(b*c))/(c*c-d*d))));
	
elif b*d>0:
	print("after multiplying");
	print(str((a*c)-(b*d))+"+i"+str((a*d)+(b*c)));	
		
	print("after dividing");
	print(str(float(((a*c)-(b*d))/(c*c-d*d)))+"+i"+str(float(((a*d)+(b*c))/(c*c-d*d))));

'''
